File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    if(n>s){
        return {-1};
    }
    vector<int> answer;
    while(n){
        answer.push_back(s/n);
        n--;
        s-=answer.back();
    }
    return answer;
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        if (n > s) {
            return new int[]{-1};
        }
        int[] answer = new int[n];
        int idx = 0;
        while (n > 0) {
            answer[idx++] = s / n;
            s -= s / n;
            n--;
        }
        return answer;
    }
}